experiment/compatibility-versions/e2e-k8s-compatibility-versions.sh

```diff
@@ -327,10 +328,16 @@ main() {
   res=0
   create_cluster || res=$?
 
-  # Clone the specific Kubernetes release branch
-  clone_kubernetes_release
+  # Clone the previous versions Kubernetes release branch
+  # TODO(aaron-prindle) extend the branches to test from n-1 -> n-1..3 as more k8s releases are done that support compatibility versions
```

Comment:
We should also think about which branches we're trying to test, if what we're trying to test is the api-server compatibility in `master`, it _might_ better suit us to use the latest stable tagged patch release in the previous branch for the other components. In practice there's not usually a big distinction anyhow though.

Reply:
Also, if we did that we could fetch binaries instead of compiling both branches from source.

Reply:
Makes sense, given that it doesn't seem to big a distinction atm perhaps we can keep this as is for now?  I created https://github.com/kubernetes/test-infra/issues/33594 to track using precompiled binaries

Reply:
Yeah, I just wanted to point that out, with the precompiled builds we have a bunch of different marker files for the subtle variations depending on what you're actually focusing on testing and how stable you want the other end of the skew.

For now 🤷, let's just get it working, but later we'll want to decide.